classes/ObjectModel.php

```diff
@@ -1468,7 +1468,13 @@ public function getCachedFieldsRequiredDatabase($all = false)
     {
         $this->cacheFieldsRequiredDatabase($all);
 
-        return $all ? self::$fieldsRequiredDatabase : self::$fieldsRequiredDatabase[$this->getObjectName()];
+        if ((bool)$all) {
+            return self::$fieldsRequiredDatabase;
+        }
+
+        return !empty(self::$fieldsRequiredDatabase[$this->getObjectName()])
+            ? self::$fieldsRequiredDatabase[$this->getObjectName()]
+            : array();
```

Comment:
What about storing `self::$fieldsRequiredDatabase[$this->getObjectName()` in a variable and then checking `!empty()` on it?
That would prevent you from calling `$this->getObjectName()` twice.